src/utils/components/SidebarEditor/SidebarEditorContext.tsx

```diff
@@ -0,0 +1,24 @@
+import React, { createContext, FC, useCallback, useState } from 'react';
+
+export type SidebarEditorContextType = {
+  showEditor: boolean;
+  setEditorVisible?: (editorVisible: boolean) => void;
+};
+
+export const SidebarEditorContext = createContext<SidebarEditorContextType>({
+  showEditor: false,
+});
+
+export const SidebarEditorProvider: FC = ({ children }) => {
+  const [showEditor, setEditor] = useState(false);
```

Comment:
usually the set funciton is exact name of the first variable, setShowEditor , also, do u really need the useCallback on line 15? i think you can simply transfer the setShowEditor itself

Reply:
Fixed